src/modelmanager.cpp

```diff
@@ -749,10 +749,35 @@ Status ModelManager::reloadModelVersions(std::shared_ptr<ovms::Model>& model, st
 
 Status ModelManager::reloadModelWithVersions(ModelConfig& config) {
     SPDLOG_LOGGER_DEBUG(modelmanager_logger, "Started applying config changes to model: {}", config.getName());
+
+    if (config.isStateful() && config.isDynamicParameterEnabled()) {
+        SPDLOG_LOGGER_ERROR(modelmanager_logger, "Requested setting dynamic parameters for stateful model {}. Dynamic shape and dynamic batch size not supported for stateful models.", config.getName());
+        return StatusCode::REQUESTED_DYNAMIC_PARAMETERS_ON_STATEFUL_MODEL;
+    }
+    if (!config.isStateful()) {
+        if (config.getMaxSequenceNumber() != DEFAULT_MAX_SEQUENCE_NUMBER) {
+            SPDLOG_LOGGER_ERROR(modelmanager_logger, "Requested setting max sequence number parameter for non stateful model {}.", config.getName());
+            return StatusCode::INVALID_NON_STATEFUL_MODEL_PARAMETER;
+        }
+        if (config.getSequenceTimeout() != DEFAULT_SEQUENCE_TIMEOUT_SECONDS) {
+            SPDLOG_LOGGER_ERROR(modelmanager_logger, "Requested setting sequencetimeout parameter for non stateful model {}.", config.getName());
+            return StatusCode::INVALID_NON_STATEFUL_MODEL_PARAMETER;
+        }
+        if (config.isLowLatencyTransformationUsed() != false) {
+            SPDLOG_LOGGER_ERROR(modelmanager_logger, "Requested low latency transformation parameter for non stateful model {}.", config.getName());
+            return StatusCode::INVALID_NON_STATEFUL_MODEL_PARAMETER;
+        }
+    }
     auto model = getModelIfExistCreateElse(config.getName(), config.isStateful());
-    if (model->isAnyVersionSubscribed() && config.isDynamicParameterEnabled()) {
-        SPDLOG_LOGGER_ERROR(modelmanager_logger, "Requested setting dynamic parameters for model {} but it is used in pipeline. Cannot reload model configuration.", config.getName());
-        return StatusCode::REQUESTED_DYNAMIC_PARAMETERS_ON_SUBSCRIBED_MODEL;
+    if (model->isAnyVersionSubscribed()) {
+        if (config.isDynamicParameterEnabled()) {
+            SPDLOG_LOGGER_ERROR(modelmanager_logger, "Requested setting dynamic parameters for model {} but it is used in pipeline. Cannot reload model configuration.", config.getName());
+            return StatusCode::REQUESTED_DYNAMIC_PARAMETERS_ON_SUBSCRIBED_MODEL;
+        }
+        if (config.isStateful()) {
```

Comment:
Beside checking here if model is used in pipeline and marked as stateful - you should as well handle this in https://github.com/openvinotoolkit/model_server/pull/489/files#diff-af2bf4798115a2c369c6337b4cc83e9498b4a41e80a5d4cbec0ce37115ac4ca0R432. 
Just check modelmanager.cpp for this retcode:
REQUESTED_DYNAMIC_PARAMETERS_ON_SUBSCRIBED_MODEL